packages/extension-ui/src/components/PasswordField/Message.tsx

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-2023 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import styled from 'styled-components';
+
+import critical from '../../assets/message-icons/critical.svg';
+import info from '../../assets/message-icons/info.svg';
+import success from '../../assets/message-icons/success.svg';
+import warning from '../../assets/message-icons/warning.svg';
+
+interface Props {
+  children: React.ReactNode;
+  className?: string;
+  messageType: 'critical' | 'warning' | 'info' | 'success';
+}
+
+const MESSAGE_TYPE_TO_ICON_URL = {
+  critical, warning, info, success
+};
+
+function Message({ children, className = '', messageType }: Props): React.ReactElement<Props> {
+  return (
+    <Container className={className}>
+      <Icon src={MESSAGE_TYPE_TO_ICON_URL[messageType]} />
+      {children}
+    </Container>
+  );
+}
+
+const Icon = styled.img`
+  width: 16px;
+  height: 16px;
+  margin-right: 5px;
+`;
+
+const Container = styled.div`
+  display: flex;
+  align-items: start;
+  line-height: 16px;
+`;
+
+export default styled(Message)<Props>`
+  color: ${({messageType, theme}) => ({critical: theme.errorColor, warning: theme.warningColor, info: theme.textColor, success: theme.primaryColor})[messageType]};
+  font-size: ${({theme}) => theme.labelFontSize};
```

Comment:
you are right, don't know what happened

Reply:
I'd actually prefer if `Container` didn't know about messageType => color mapping. I think it's a different level of abstraction. If you don't mind I'll leave `color` where it is and move `font-size`.

Reply:
Imho it's still fine for the `<Container>` to know it, because it's _the `<Message>`'s container_, so it plays a role in the overal `<Message>` appearance / logic.

Alternatively, you could also have the `color` property set through the regular style prop (`<Container style={{ color: mapping }}>`, which is imo as well perfectly fine since we want `color` to be unoverridable in this case anyway and creating a class (by SC under the hood) for each possible color seems kind of opposite to the classes purpose.

At the end of the day it's up to you - I'm not fixed on any of those solutions in this case, just providing options.

Reply:
I don't mind either solution, but I like the current one - the `Message` component is pure - doesn't use theme - and the color is injected through styled components, which know about theme anyway. I find it aesthetic. Then again, I don't mind either solution, so let me know what you think :sweat_smile:

Reply:
That's fine, let's leave it then.